borrowing_service/views.py

```diff
@@ -1,3 +1,69 @@
-from django.shortcuts import render
+import datetime
 
-# Create your views here.
+from rest_framework import viewsets, mixins, status
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from borrowing_service.models import Borrowing
+from borrowing_service.serializers import (
+    BorrowingSerializer,
+    BorrowingDetailSerializer,
+    BorrowingListSerializer
+)
+
+
+class BorrowingViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Borrowing.objects.all()
+    serializer_class = BorrowingSerializer
+    permission_classes = [IsAuthenticated,]
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if is_active := self.request.query_params.get("is_active", False):
+            queryset = queryset.filter(actual_return_date__isnull=True)
+
+        if self.request.user.is_staff:
+
+            if user_id := self.request.query_params.get("user_id"):
+                queryset = queryset.filter(user_id=user_id)

```

Comment:
You no need to use this logic. You can get user from your request and make queryset filter with it.